internal/cli/uploadcmd/upload.go

```diff
@@ -89,6 +104,7 @@ func New(config *rootcmd.Config) *ffcli.Command {
 	fs.StringVar(&cfg.S3Region, "s3-region", "", "Region of the S3 bucket when using the S3 registry type")
 	fs.StringVar(&cfg.GCSBucket, "gcs-bucket", "", "Bucket to use when using the GCS registry type")
 	fs.StringVar(&cfg.GCSPrefix, "gcs-prefix", "", "Prefix to use when using the GCS registry type")
+	fs.StringVar(&flagVersionConstraints, "version-constraints", "", "Limit the module versions that are eligible for upload with version constraints. The version string has to be formatted as a string literal containing one or more conditions, which are separated by commas")
```

Comment:
Can we put this in the struct as well to be consistent with the other flags? 🙂 

```go
fs.StringVar(&cfg.VersionConstraints, ...)
```

Reply:
That's what I did initially, but it didn't feel right to put a "meaningless" `string` in a struct. As essentially what we care about is the parsed and validated `version.Constraints`. So from an architecture perspective, that's what I want to pass on.
But I get your point, I spent a lot of time thinking which way I prefer. And how I did it doesn't feel correct as well, so I'm happy to change that :)